src/CMakeLists.txt

```diff
@@ -345,20 +417,26 @@ if(CMAKE_CROSSCOMPILING)
   endforeach(firmware)
 
   target_sources(firmware.elf PRIVATE firmware.c)
-  target_compile_definitions(firmware.elf PRIVATE APP_BTC APP_LTC APP_ETH APP_U2F)
+  target_compile_definitions(firmware.elf PRIVATE FIRMWARE_MULTI APP_BTC APP_LTC APP_ETH APP_U2F)
   target_sources(firmware.elf PRIVATE ${FIRMWARE-U2F-SOURCES})
 
   target_sources(firmware-semihosting.elf PRIVATE firmware.c)
   target_sources(firmware-semihosting.elf PRIVATE ${FIRMWARE-U2F-SOURCES})
   # Select an implementation of the system calls that can communicate with the debugger
   target_compile_options(firmware-semihosting.elf PRIVATE --specs=rdimon.specs)
   target_link_libraries(firmware-semihosting.elf PRIVATE --specs=rdimon.specs)
-  target_compile_definitions(firmware-semihosting.elf PRIVATE APP_BTC APP_LTC APP_ETH APP_U2F)
+  target_compile_definitions(firmware-semihosting.elf PRIVATE FIRMWARE_MULTI APP_BTC APP_LTC APP_ETH APP_U2F)
   target_compile_definitions(firmware-semihosting.elf PRIVATE SEMIHOSTING)
 
   target_sources(firmware-btc.elf PRIVATE firmware.c)
   target_compile_definitions(firmware-btc.elf PRIVATE FIRMWARE_BTC_ONLY APP_BTC)
 
+  target_sources(firmware-base.elf PRIVATE bitboxbase/bitboxbase.c commander/commander_bitboxbase.c)
+  target_compile_definitions(firmware-base.elf PRIVATE FIRMWARE_BITBOXBASE APP_BTC)
```

Comment:
I'd also add `APP_BITBOXBASE`, and then use that to include/exclude the bitboxbase api calls (similar to ETH).

That would be more consistent with how we do it with the other editions.

Reply:
I would differentiate "platform" from "app". Maybe `PLATFORM_BITBOXBASE`?

Reply:
Maybe we should remove all `FIRMWARE_` prefixed defines and use `TARGET_BITBOXMULTI`, `TARGET_BITBOXBTCONLY` and `TARGET_BITBOXBASE`.

Reply:
the base is an edition on the same level as multi, btconly, right?

i was just saying that the base api calls could be in the `bitboxbase` app, to be consistent with rest.

e.g. [here](https://github.com/digitalbitbox/bitbox02-firmware/pull/162/files#diff-82b64e48d7dd3ab4dbc37ae5d9ff3dceR310) it is also `#if defined(APP_ETH)`, not `#if defined(FIRMWARE_MULTI)`. The `#if defined(FIRMWARE_BITBOXBASE)` a few lines below is kind of breaking the pattern.

Not a big deal though.

Reply:
Yeah, but I'm just saying that we have two types of abstractions right now, firmware targets such as multi/bitcoin-only/bitboxbase and that we have apps such as BTC, ETH, U2F, perhaps "autosigning with whitelisting". And I agree that calling it `FIRMWARE_BITBOXBASE` might be misleading, but calling it `APP_BITBOXBASE` would be completely wrong IMO.